.circleci/config.yml

```diff
@@ -241,73 +233,42 @@ workflows:
   test_all_python_versions:
     jobs:
       - win_unit_tests:
-          name: "py36 windows unit tests"
-          python_version: "3.6"
-      - win_unit_tests:
-          name: "py37 windows unit tests"
-          python_version: "3.7"
-      - win_unit_tests:
-          name: "py38 windows unit tests"
-          python_version: "3.8"
-      - install_ft:
-          name: "py36 install featuretools"
-          image_tag: "3.6"
+          matrix:
+            parameters:
+              python_version: ["3.8", "3.7", "3.6"]
+          name: py<< matrix.python_version >> windows unit tests
       - install_ft:
-          name: "py37 install featuretools"
-      - install_ft:
-            name: "py38 install featuretools"
-            image_tag: "3.8"
+          matrix:
+            parameters:
+              image_tag: ["3.8", "3.7", "3.6"]
+          name: << matrix.image_tag >> install featuretools
       - unit_tests:
-          name: "py36 unit tests"
-          image_tag: "3.6"
-          codecov: true
-          requires:
-            - "py36 install featuretools"
-      - unit_tests:
-          name: "py37 unit tests"
-          requires:
-            - "py37 install featuretools"
-      - unit_tests:
-          name: "py38 unit tests"
-          image_tag: "3.8"
-          requires:
-            - "py38 install featuretools"
-      - lint_test:
-          name: "py36 lint test"
-          image_tag: "3.6"
-          requires:
-            - "py36 install featuretools"
-      - lint_test:
-          name: "py37 lint test"
+          matrix:
+            parameters:
+              image_tag: ["3.8", "3.7", "3.6"]
+              codecov: [false, false, true]
```

Comment:
I think this is going to cause 9 tests to be launched - one for each combination of image_tag and codecov.

Reply:
Maybe we can remove the `codecov` parameter here and change the `when` condition in the `lint_test` job to use the python version instead, and launch when we are running 3.6.

Reply:
Fixed